### PR TITLE
[subset] fix for name table serializing

### PR DESCRIPTION
**Pull request by qxliu76**

### CODE REVIEW

File: src/hb-null.hh [hunk 1]
```diff
@@ -67,7 +67,7 @@ template <typename T, typename B>
 struct _hb_static_size
 { enum { value = sizeof (T) }; };
 template <typename T>
-struct _hb_static_size<T, hb_bool_tt<true || sizeof (T::min_size)>>
+struct _hb_static_size<T, hb_bool_tt<true || sizeof (T::min_size)> >
```
Comment (behdad) [inline src/hb-null.hh]: These are unrelated and undesired changes.  Please remove.

File: src/hb-ot-name-table.hh [hunk 2]
```diff
@@ -100,7 +100,9 @@ struct NameRecord
 		    const void *dst_base) const
   {
     auto *out = c->embed (this);
-    out->offset = 0;
+    if (!out)
+      return nullptr;
+
```
Comment (behdad) [inline src/hb-ot-name-table.hh]: Instead of the three lines, just have one line:
```
if (unlikely (!out)) return_trace (false);
```
and yes, add tracing to the function please.

File: src/hb-ot-name-table.hh [hunk 3]
```diff
@@ -292,6 +294,9 @@ struct name
     + hb_iter (name_record_idx_to_retain)
     | hb_apply ([&] (unsigned _) { c->copy (src_array[_], src_base, dst_base); })
     ;
+ 
+    if (c->ran_out_of_room)
+      return_trace (false);
```
Comment (behdad) [inline src/hb-ot-name-table.hh]: Add `unlikely`, and return can go on the same line.

**Comment by qxliu76** [comment]
> Thanks. Do the tests pass with this, or that's a separate issue?

this is fix for failed tests.

**Comment by behdad** [comment]
Ah, so the assert was failing?

**Comment by qxliu76** [comment]
the tests are failing in NameRecord->copy (), in which c->embed() returns a nullptr due to ran_out_of_room. Then test crashesdue to serialize_copy with a nullptr.
